test/gtest/dropout_ocl_hip.cpp

```diff
@@ -0,0 +1,1068 @@
+                    bool use_rsvsp,
+                    PerfHelper<float>& perf_helper,
+                    bool use_hip)
+{
+
+    auto&& handle = get_handle();
+    auto din_gpu  = din;
+
+    auto din_dev   = handle.Write(din.data);
+    auto dout_dev  = handle.Write(dout.data);
+    auto rsvsp_dev = handle.Write(rsvsp);
+
+    DropoutBackward(handle,
+                    din.desc,
+                    dout.desc,
+                    dout_dev.get(),
+                    din.desc,
+                    din_dev.get(),
+                    use_rsvsp ? rsvsp_dev.get() : nullptr,
+                    rsvsp.size(),
+                    in_offset,
+                    out_offset,
+                    rsvsp_offset,
+                    DropoutDesc,
+                    perf_helper,
+                    use_hip);
+
+    din_gpu.data = handle.Read<T>(din_dev, din.data.size());
+    return din_gpu;
+}
+
+struct DropoutTestCase
+{
+    size_t dim0;
+    size_t dim1;
+    size_t dim2;
+    size_t dim3;
+    size_t dim4;
+
+    friend std::ostream& operator<<(std::ostream& os, const DropoutTestCase& tc)
+    {
+        return os << " dim0:" << tc.dim0 << " dim1:" << tc.dim1 << " dim2:" << tc.dim2
+                  << " dim3:" << tc.dim3 << " dim4:" << tc.dim4;
+    }
+
+    std::vector<size_t> GetInput()
+    {
+        // check if the tensor is 1D, 2D, 3D, 4D or 5D
+        if(dim4 > 1)
+            return {dim0, dim1, dim2, dim3, dim4};
+        else if(dim3 > 1)
+            return {dim0, dim1, dim2, dim3};
+        else if(dim2 > 1)
+            return {dim0, dim1, dim2};
+        else if(dim1 > 1)
+            return {dim0, dim1};
+        else
+            return {dim0};
+    }
+};
+
+std::vector<DropoutTestCase> DropoutTestConfigs()
+{
+
+    if constexpr(PERF_ENABLE)
+    {
+
+        std::vector<DropoutTestCase> configs;
+        const auto& handle = get_handle();
+        size_t maxTotalSize;
+
+        // Generate all NCHW tensors that are limited by L3 cache size
+        // or 2xL2 cache size when L3 is not available
+        if(miopen::StartsWith(handle.GetDeviceName(), "gfx90a") ||
+           miopen::StartsWith(handle.GetDeviceName(), "gfx908"))
+        {
+            maxTotalSize = 16; // 8MB L2
+        }
+        else if(miopen::StartsWith(handle.GetDeviceName(), "gfx803"))
+        {
+            maxTotalSize = 4; // 2MB L2
+        }
+        else if(miopen::StartsWith(handle.GetDeviceName(), "gfx900") ||
+                miopen::StartsWith(handle.GetDeviceName(), "gfx906"))
+        {
+            maxTotalSize = 8; // 4MB L2
+        }
+        else if(miopen::StartsWith(handle.GetDeviceName(), "gfx942"))
+        {
+            maxTotalSize = 256; // 128MB L3
+        }
+        else if(miopen::StartsWith(handle.GetDeviceName(), "gfx103"))
+        {
+            maxTotalSize = 128; // 128MB L3
+        }
+        else
+        {
+            maxTotalSize = 4; // 2MB L2, default case.
+        }
```

Comment:
Cache size is in megabytes, not elements.
So proper `maxTotalSize` should be like this: `maxTotalSize *= 1024ull * 1024ull;` in megabytes or `maxTotalSize *= 1024ull * 1024ull / sizeof(DataType);` in elements.

Also I would recommend making more clear comment:
```
maxTotalSize = 16; // twice than 8MB L2
```